packages/hathor-rpc-handler/src/helpers/transactions.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { HathorWallet } from '@hathor/wallet-lib';
+import { HATHOR_TOKEN_CONFIG } from '@hathor/wallet-lib/lib/constants';
+import { QueryUtxosFilters, SendTxInput, SendTxInputSpecific, SendTxOutput, UtxoDetails, UtxoInfo } from '../types';
+import { NoUtxosAvailableError } from '../errors';
+
+export interface TokenOutput {
+  tokenUid: string;
+  amount: number;
+}
+
+export interface PreparedInput {
+  txId: string;
+  index: number;
+}
+
+export interface PreparedTxResponse {
+  success: boolean;
+  outputs: SendTxOutput[];
+  inputs: PreparedInput[];
+}
+
+export async function getUtxosToFillTx(
+  wallet: HathorWallet,
+  sumOutputs: number,
+  options: QueryUtxosFilters,
+): Promise<UtxoInfo[] | null> {
+  // We want to find only utxos to use in the tx, so we must filter by available only
+  const getUtxosOptions = {
+    ...options,
+    only_available_utxos: true
+  };
+  const utxosDetails: UtxoDetails = await wallet.getUtxos(getUtxosOptions);
```

Comment:
If the idea is to select utxos to fill a transaction, you should use the method `bestUtxoSelection` from utxo utils.

It will remove most of the code you have here.